src/Storages/transformQueryForExternalDatabaseAnalyzer.cpp

```diff
@@ -76,7 +93,13 @@ ASTPtr getASTForExternalDatabaseFromQueryTree(const QueryTreeNodePtr & query_tre
     if (union_ast->list_of_selects->children.size() != 1)
         throw Exception(ErrorCodes::UNSUPPORTED_METHOD, "QueryNode AST is not a single ASTSelectQuery, got {}", union_ast->list_of_selects->children.size());
 
-    return union_ast->list_of_selects->children.at(0);
+    ASTPtr select_query = union_ast->list_of_selects->children.at(0);
+    auto * select_query_typed = select_query->as<ASTSelectQuery>();
+    if (!select_query_typed)
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Expected ASTSelectQuery, got {}", select_query ? select_query->formatForErrorMessage() : "nullptr");
+    if (!allow_where)
+        select_query_typed->setExpression(ASTSelectQuery::Expression::WHERE, nullptr);
```

Comment:
Probably, it's more correct to modify the WHERE filter expression, because it can contain references to both tables. We should leave only conjunctions that are related to the proper table.

Reply:
It works like that now, but for OUTER join, we don't want to keep filters for related table either.

```sql

SELECT * FROM ( SELECT arrayJoin([1, 2]) AS a ) AS t1
FULL OUTER JOIN ( SELECT arrayJoin([1, 2]) AS a ) AS t2 ON t1.a = t2.a
WHERE t1.a != 1

   ┌─a─┬─t2.a─┐
1. │ 2 │    2 │
   └───┴──────┘
   
SELECT * FROM ( SELECT arrayJoin([1, 2]) AS a WHERE a != 1 ) AS t1
FULL OUTER JOIN ( SELECT arrayJoin([1, 2]) AS a ) AS t2 ON t1.a = t2.a
WHERE t1.a != 1

   ┌─a─┬─t2.a─┐
1. │ 2 │    2 │
2. │ 0 │    1 │
   └───┴──────┘
```